include/ginkgo/core/preconditioner/isai.hpp

```diff
@@ -122,10 +125,11 @@ class Isai : public EnableLinOp<Isai<IsaiType, ValueType, IndexType>>,
                        ? isai_type::spd
                        : IsaiType == isai_type::lower ? isai_type::upper
                                                       : isai_type::lower,
-             ValueType, IndexType>;
+             ValueType, IndexType, StorageType>;
```

Comment:
`transposed_type` can be simplified by only having special treatment for `lower` and `upper`, while all others remain the same.